ultravox/tools/tts.py

```diff
@@ -145,9 +147,38 @@ def tts(self, text: str, voice: Optional[str] = None):
         return self._handle_pcm_response(self._post(url, headers, body))
 
 
+class CachedClientWrapper:
+    def __init__(self, client: Client, provider: str):
+        super().__init__()
+        self._client = client
+        self._base_path = os.path.join(".cache/ds_tool", provider)
+
+    def tts(self, text: str, voice: Optional[str] = None):
+        path = os.path.join(self._base_path, voice or "default")
+        text_hash = hashlib.sha256(text.encode()).hexdigest()
+        os.makedirs(path, exist_ok=True)
+
+        cache_path = os.path.join(path, f"{text_hash}.wav")
+
+        if os.path.exists(cache_path):
+            with open(cache_path, "rb") as f:
+                return f.read()
+
+        pcm = self._client.tts(text, voice)
```

Comment:
Output is actually WAV rather than PCM, FWIW

Reply:
It shows that I still don't know what PCM is, no? :))
Renaming to wav.